### ISSUE TITLE
VCF CLI questions

### ISSUE THREAD

**Issue by trvrb** [nextstrain/augur#153]
@emmahodcroft ---

I spent some more time with the interface and I was left with a couple questions regarding VCF implementation.

1. Should `augur tree --strip-sites` take a BED file like `augur mask` does? This would seem more consistent. They do basically the same thing. Might as well stick with standard file format.

2. Is there ever really a reason not to include `--keep-vcf-fasta`? I would set this to always be present. It's like `location.mugration_model.txt`. Useful auxiliary output. And in terms of consistency, we never ask if we should `--keep-mugration-model-txt`.

**Comment by emmahodcroft** [comment]
1. Yes, we could switch it to a BED file. At the moment, this is intended for single bp DRM sites (which may only apply for HIV and TB), so the BED format is a bit overkill (don't need either 'Chrom' or 'ChromEnd' which are 2 of the 3 required fields). BED format would be useful for excluding larger regions from topology-building, but I'm not sure this makes sense when they'll still be used in `refine` and impact branch lengths. However, it wouldn't be hard to implement.
I was hoping to find a standardised DRM file format to use both for this and the DRM identification step (not yet implemented in new augur), but haven't found such a thing yet...

2. I agree

**Comment by trvrb** [comment]
Thanks @emmahodcroft. I just removed `--keep-vcf-fasta` in commit 50c7b458f2f6e297ccebc2593892fd503fde65e8.

I actually had a follow up question on 1. I think that you want to still mask these sites when running `augur tree` and also `augur refine`, ie you don't want these sites screwing with the clock. In this case, I would suggest to just do everything in `augur mask` and remove the `augur tree --strip-sites` option. In this case, you'd run:
```
augur mask -a filtered.fasta -o masked.fasta
augur tree -a masked.fasta -o tree_raw.nwk
augur refine -t tree_raw.nwk -a masked.fasta -output-tree tree.nwk
augur ancestral -t tree.nwk -a filtered.fasta
```
This would reconstruct all sites in the alignment, but the tree structure and dating would only be from good sites. Generally, seems like less overloading to have `augur mask` do the masking rather than also having an internal mask in `augur tree`. But if you think better to keep `augur tree --strip-sites`, I'm fine with that as well.

Fairly agnostic between simple site-wise format and BED, but I do think that if they both exist `augur mask` and `augur tree --strip-sites` should both be able to at least take the same file formats. This might mean taking multiple options (as I can see why you'd maybe want the same list of DRM sites to mask as to annotate with `augur find-drm`).

**Comment by emmahodcroft** [comment]
In the case of HIV and TB I think including the sites in `refine` is probably ok, as these mutations do arise genuinely (not through recombination) (though tbh, they probably have little effect on the clock anyway). However, certainly this is specific to 1bp DRM sites, and wouldn't be what you wanted for larger sections of a genome, which this isn't intended for. 

The downside of using the unmasked alignment for `ancestral` is that it's going to put mutations on the tree that weren't used in `tree` or `refine` (many more than just the DRM positions), so could make the final output quite confusing.

So while it's a bit of a special case, I do think it's worth keeping separate with `--strip-sites`. 

I agree that probably the best route is to accept multiple file formats, and that shouldn't be difficult; I'll work on it now.

**Comment by trvrb** [comment]
This makes sense. Thanks @emmahodcroft.

**Comment by huddlej** [comment]
This was closed by #157
